fastparquet/converted_types.py

```diff
@@ -119,21 +119,13 @@ def from_bytes(d):
     elif ctype == parquet_thrift.ConvertedType.DATE:
         return (data * DAYS_TO_MILLIS).view('datetime64[ns]')
     elif ctype == parquet_thrift.ConvertedType.TIME_MILLIS:
-        out = np.empty(len(data), dtype='int64')
-        time_shift(data, out, 1000000)
-        return out.view('timedelta64[ns]')
+        return time_shift(data, 1000000).view('timedelta64[ns]')
```

Comment:
we put all the datatype conversion and array allocation into the `time_shift` function